ddl/ddl_api.go

```diff
@@ -3549,6 +3599,16 @@ func (d *ddl) DropIndex(ctx sessionctx.Context, ti ast.Ident, indexName model.CI
 	}
 
 	indexInfo := t.Meta().FindIndexByName(indexName.L)
+	if isPK {
+		if !config.GetGlobalConfig().AlterPrimaryKey || t.Meta().PKIsHandle {
+			return ErrUnsupportedModifyPrimaryKey.GenWithStack(fmt.Sprintf(
+				"unsupported drop primary key, alter-primary-key is %v, the table's pk-is-handle is %v",
+				config.GetGlobalConfig().AlterPrimaryKey, t.Meta().PKIsHandle))
+		}
+		if indexInfo == nil {
+			return ErrCantDropFieldOrKey.GenWithStack("Can't DROP 'PRIMARY'; check that column/key exists")
```

Comment:
```suggestion
			return ErrCantDropFieldOrKey.GenWithStack("index `PRIMARY` doesn't exist")
```
I think it is enough.

Reply:
I refer to the MySQL error message：
```
mysql> alter table t4 drop primary key;
ERROR 1091 (42000): Can't DROP 'PRIMARY'; check that column/key exists
```

Reply:
got it